UnitsNet.Tests/CustomCode/TemperatureTests.cs

```diff
@@ -27,106 +25,103 @@ public class TemperatureTests : TemperatureTestsBase
 
         protected override double KelvinsInOneKelvin => 1;
 
+        public static IEnumerable<object[]> DividedByTemperatureDeltaEqualsTemperatureData { get; } =
+            new List<object[]>
+            {
+                new object[] { Temperature.FromDegreesCelsius(10), 1, Temperature.FromDegreesCelsius(10) },
+                new object[] { Temperature.FromDegreesCelsius(10), 5, Temperature.FromDegreesCelsius(2) },
+                new object[] { Temperature.FromDegreesCelsius(10), -10, Temperature.FromDegreesCelsius(-1) },
+                new object[] { Temperature.FromDegreesFahrenheit(10), 1, Temperature.FromDegreesFahrenheit(10) },
+                new object[] { Temperature.FromDegreesFahrenheit(10), 5, Temperature.FromDegreesFahrenheit(2) },
+                new object[] { Temperature.FromDegreesFahrenheit(10), -10, Temperature.FromDegreesFahrenheit(-1) }
+            };
+
         [SuppressMessage("ReSharper", "ImpureMethodCallOnReadonlyValueField",
             Justification = "R# incorrectly identifies method as impure, due to internal method calls.")]
         [Theory]
-        [InlineData(TemperatureUnit.DegreeCelsius, 10, 1, "10 °C")]
-        [InlineData(TemperatureUnit.DegreeCelsius, 10, 5, "2 °C")]
-        [InlineData(TemperatureUnit.DegreeCelsius, 10, -10, "-1 °C")]
-        [InlineData(TemperatureUnit.DegreeFahrenheit, 10, 1, "10 °F")]
-        [InlineData(TemperatureUnit.DegreeFahrenheit, 10, 5, "2 °F")]
-        [InlineData(TemperatureUnit.DegreeFahrenheit, 10, -10, "-1 °F")]
-        public void DividedByTemperatureDeltaEqualsTemperature(TemperatureUnit unit, int temperatureVal, int divisor, string expected)
+        [MemberData(nameof(DividedByTemperatureDeltaEqualsTemperatureData))]
+        public void DividedByTemperatureDeltaEqualsTemperature(Temperature temperature, int divisor, Temperature expected)
         {
-            Temperature temperature = Temperature.From(temperatureVal, unit);
-
-            // Act
-            Temperature resultTemp = temperature.Divide(divisor, unit);
-
-            var resultTempInUnit = resultTemp.ToUnit(unit);
-            var actual = string.Format(CultureInfo.InvariantCulture, "{0:0} {1:a}", resultTempInUnit.Value, resultTempInUnit);
-            Assert.Equal(expected, actual);
+            Temperature resultTemp = temperature.Divide(divisor, temperature.Unit);
+            Assert.True(expected.Equals(resultTemp, 1e-5, ComparisonType.Absolute));
```

Comment:
Shouldn't use our implementation of `Equals` here since that would be testing one implementation with another implementation. Better to just write the comparison check manually as I don't think Xunit has good support for it (I may be wrong). Something like this, can be added as a util method for test code.
`Assert.True(Math.Abs(a-b) <= 1e-5);`

Same thing in other tests below.

Reply:
I don't think it's completely avoidable. Other tests would fail too (Equals tests) if it was wrong. All tests rely on the code to some degree - even ```Temperature.Fromxxx```

Reply:
I suppose you are right to some extent it is unavoidable. Equals could be avoidable though, but I agree with your reasoning that Equals is already tested and should be fine.